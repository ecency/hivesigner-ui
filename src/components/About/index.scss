.AboutContainer {
  display: flex;
  flex-direction: column;
  margin-top: 10vh;
  transition: all 0.8s;
}
.Logo {
  display: flex;
  flex-direction: column;
  align-items: center;
  .about_logo {
    width: 70px;
  }
  span {
    color: red;
    font-size: 2.8rem;
    font-weight: 500;
  }
}
.AboutContent {
  margin-top: 3rem;
  display: flex;
  flex-direction: column;
  transition: all 0.8s;
  span {
    margin-top: 15px;
    margin-bottom: 15px;
    font-size: 1.15rem;
    color: rgba(0, 0, 0, 0.9);
  }
  @media screen and (max-width: 600px) {
    margin: 10px;
  }
}
.Title {
  font-weight: 600;
  color: black !important;
  font-size: 2.5rem !important;
  text-align: left;
}
.Detail {
  margin-top: 20px;
  font-size: 1.25rem;
  color: rgba(0, 0, 0, 0.9);
  max-width: 550px;
}
.LinkGroup {
  display: flex;
  flex-direction: column;
  a {
    display: flex;
    flex-direction: row;
    align-items: center;
    grid-gap: 10px;
    margin-bottom: 15px;
  }
  transition: all 0.8s;
  @media screen and (max-width: 600px) {
    margin: 10px;
  }
}
.Contributors {
  display: flex;
  flex-direction: column;
  font-size: 1.15rem;
  transition: all 0.8s;
  @media screen and (max-width: 600px) {
    margin: 10px;
  }
  * {
    margin-top: 10px;
  }
  span {
    color: black;
  }
  color: rgba(0, 0, 0, 0.8);
  span a {
    color: rgb(0, 110, 255);
  }
  a:hover {
    text-decoration: underline;
  }
  span a:hover {
    text-decoration: underline;
  }
  margin-bottom: 50px;
}
.AboutMenubar {
  font-size: 2rem;
  color: black;
  position: absolute;
  right: 30px;
  top: 50px;
  display: block;
  @media screen and (min-width: 630px) {
    display: none;
  }
}
.Aboutfix {
  * {
    top: 30px !important;
  }
}
