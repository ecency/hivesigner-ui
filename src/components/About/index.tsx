import React, { useState, useEffect } from "react";
import { Link } from "react-router-dom";
import { AiOutlineSelect, AiFillGithub } from "react-icons/ai";
import Content from "../../layouts/content";
import "./index.scss";
import HomeLogo from "../../assets/img/logo";
import { ImMenu } from "react-icons/im";
import { AiOutlineClose } from "react-icons/ai";
import Navbar from "../Index/Navbar";
const Index = () => {
  const [showNavbar, setshowNavbar] = useState(true);
  const handleChange = () => {
    setshowNavbar(!showNavbar);
  };
  useEffect(() => {
    let window_width = window.matchMedia("(min-width:630px)");
    window_width.addListener(handleScreenChange);
    handleScreenChange(window_width);
    return () => {
      window_width.removeListener(handleScreenChange);
    };
  }, []);
  const handleScreenChange = (mediaQuery: any) => {
    if (mediaQuery.matches) {
      setshowNavbar(true);
    } else {
      setshowNavbar(false);
    }
  };
  return (
    <Content>
      <div className="AboutContainer">
        <div className="AboutMenubar" onClick={() => handleChange()}>
          {showNavbar ? <AiOutlineClose /> : <ImMenu />}
        </div>
        <div className="Logo">
          {/* <img
            draggable="false"
            width="70px"
            height="auto"
            src={HomeLogo}
            alt="logo"
          /> */}
          <div className="about_logo">
            <HomeLogo />
          </div>
          <span>About</span>
        </div>
        <div className="AboutContent">
          <div className="Title">Hivesigner</div>
          <div className="Detail">
            Secure way to sign with Hivesigner. Best security for users and
            developers to integrate industry standard OAuth2 for their
            Blockchain applications. Transform web 2.0 apps into web 3.0
            decentralized apps.
          </div>
          <span>Version:1.0.1</span>
          <span>License:MIT</span>
        </div>
        <div className="LinkGroup">
          <Link to="/">
            <AiOutlineSelect />
            Website
          </Link>
          <Link to="/">
            <AiOutlineSelect />
            Download Logo
          </Link>
          <a href="https://github.com/ecency/hivesigner-ui/issues">
            <AiFillGithub />
            Report a bug
          </a>
        </div>
        <div className="Contributors">
          <span>Contributors</span>
          <a href="https://ecency.com/@good-karma">Feruz Muradov</a>
          <a href="https://ecency.com/@dkildar">Ildar Timerbaev</a>
          <a href="https://ecency.com/@zetrix">Dmytro Khomenchuk</a>
          <a href="https://ecency.com/@fabien">Fabien Marino</a>
          <a href="https://ecency.com/@sekhmet">Wiktor Tkaczyński</a>
          <a href="https://ecency.com/@jnordberg">Johan Nordberg</a>
          <a href="https://ecency.com/@wehmoen">Nico Wehmöller</a>
          <a href="https://ecency.com/@mahdiyari">Mahdi Yari</a>
          <span>
            Maintained by<a href="https://ecency.com/">Ecency Team .</a>
          </span>
        </div>
        {showNavbar && (
          <div className="Aboutfix">
            <Navbar />
          </div>
        )}
      </div>
    </Content>
  );
};

export default Index;
