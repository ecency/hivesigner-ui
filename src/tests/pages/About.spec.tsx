import React from "react";
import About from "../../components/About";
import renderer from "react-test-renderer";
describe("About component Test", () => {
  test("UI Test", () => {
    const about = renderer.create(<About />);
    let tree = about.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
