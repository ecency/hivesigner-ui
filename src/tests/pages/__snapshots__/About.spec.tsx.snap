// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`About component Test 1 Snapshot testing 1`] = `
<div
  className="AboutContainer"
/>
`;
